tenprintcover.py

```diff
@@ -498,25 +498,50 @@ def drawShape(c, x, y, s):
         else:
             assert not "Implement."
 
-
+    # If the text is long, use a smaller font size. 
+    def scale_font(text, font_name, font_properties):
+        (font_size, font_slant, font_weight) = font_properties
+        width = len(text) * font_size
+        if width > cover_width * 7:   # This is an empirical, unintelligent, heuristic.
+            return  (font_size * 0.8, font_slant, font_weight)
+        else:
+            return font_properties
+    
+    # Return a font appropriate for the text. Uses Noto CJK if text contains letters of
+    # Simplified Chinese, Traditional Chinese, Japanese, and Korean (CJK), otherwise Noto Sans. 
+    # http://www.unicode.org/faq/han_cjk.html
+    def select_font(text):
+        for char in text:
+            if ord(char) >= 0x4E00:
+                return 'Noto Sans CJK SC'
+        return 'Noto Sans'
```

Comment:
Instead of a manual iteration and a hardwired check against the [16b Unihan constant `4E00`](http://unicode.org/cgi-bin/GetUnihanData.pl?codepoint=4e00), I would like to propose using the [regex](https://pypi.python.org/pypi/regex) module which provides support for [Unicode character properties](https://en.wikipedia.org/wiki/Unicode_character_property#Script):

```python
import regex

# Pre-compile regular expression for later: search for characters covered
# by the Han script; this is easier than specifying the blocks which are
# covered by the script.
_re = regex.compile(r"\p{Script=Han}+")

def select_font(text):
    return "Noto Sans" if _re.search(text) is None else "Noto Sans CJK SC"
```

We’d probably have to extend the script though, to [Hrkt (or Kana)](https://en.wikipedia.org/wiki/Kana) for the Japanese scripts, and to [Kore (Han + Hangul)](https://en.wikipedia.org/wiki/Korean_mixed_script) for the Korean.

```python
_re = regex.compile(r"(\p{Script=Han}\p{Script=Hrkt}\p{Script=Kore})+")
```